File: android/src/newarch/java/com/hector/nativewechat/NativeWechatModule.java
package com.hector.nativewechat;

import com.facebook.react.bridge.Callback;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReadableMap;
import com.facebook.react.module.annotations.ReactModule;

@ReactModule(name = NativeWechatModule.NAME)
public class NativeWechatModule extends NativeWechatSpec {
  public static final String NAME = "Wechat";
  private NativeWechatModuleImpl moduleImpl;

  public NativeWechatModule(ReactApplicationContext context) {
    super(context);
    moduleImpl = new NativeWechatModuleImpl(context);
  }

  @Override
  public void sendAuthRequest(ReadableMap request, Callback callback) {
    moduleImpl.sendAuthRequest(request, callback);
  }

  @Override
  public void registerApp(ReadableMap request) {
    moduleImpl.registerApp(request);
  }

  @Override
  public void shareText(ReadableMap request, Callback callback) {
    moduleImpl.shareText(request, callback);
  }

  @Override
  public void shareImage(ReadableMap request, Callback callback) {
    moduleImpl.shareImage(request, callback);
  }

  @Override
  public void shareVideo(ReadableMap request, Callback callback) {
    moduleImpl.shareVideo(request, callback);
  }

  @Override
  public void shareWebpage(ReadableMap request, Callback callback) {
    moduleImpl.shareWebpage(request, callback);
  }

  @Override
  public void shareMiniProgram(ReadableMap request, Callback callback) {
    moduleImpl.shareMiniProgram(request, callback);
  }

  @Override
  public void isWechatInstalled(Callback callback) {
    moduleImpl.isWechatInstalled(callback);
  }

  @Override
  public void requestPayment(ReadableMap request, Callback callback) {
    moduleImpl.requestPayment(request, callback);
  }

  @Override
  public void requestSubscribeMessage(ReadableMap request, Callback callback) {
    moduleImpl.requestSubscribeMessage(request, callback);
  }

  @Override
  public void launchMiniProgram(ReadableMap request, Callback callback) {
    moduleImpl.launchMiniProgram(request, callback);
  }

  @Override
  public void openCustomerService(ReadableMap request, Callback callback) {
    moduleImpl.openCustomerService(request, callback);
  }

  @Override
  public void addListener(String eventName) {
    // 保持空实现，只是为了满足接口要求
  }
  
  @Override
  public void removeListeners(double count) {
    // 保持空实现，只是为了满足接口要求
  }
}

File: android/src/main/java/com/hector/nativewechat/NativeWechatModule.kt
package com.hector.nativewechat

import com.facebook.react.bridge.Callback
import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReadableMap
import com.facebook.react.bridge.ReactMethod
import com.facebook.react.bridge.WritableMap

class NativeWechatModule internal constructor(context: ReactApplicationContext) : WechatSpec(context) {
    private val moduleImpl = NativeWechat(context)

    override fun getName(): String = NAME

    // getConstants 已经在 WechatSpec 中通过 getTypedExportedConstants 实现

    @ReactMethod
    override fun sendAuthRequest(request: ReadableMap, callback: Callback) {
        moduleImpl.sendAuthRequest(request, callback)
    }

    @ReactMethod
    override fun registerApp(request: ReadableMap) {
        moduleImpl.registerApp(request)
    }

    @ReactMethod
    override fun shareText(request: ReadableMap, callback: Callback) {
        moduleImpl.shareText(request, callback)
    }

    @ReactMethod
    override fun shareImage(request: ReadableMap, callback: Callback) {
        moduleImpl.shareImage(request, callback)
    }

    @ReactMethod
    override fun shareVideo(request: ReadableMap, callback: Callback) {
        moduleImpl.shareVideo(request, callback)
    }

    @ReactMethod
    override fun shareWebpage(request: ReadableMap, callback: Callback) {
        moduleImpl.shareWebpage(request, callback)
    }

    @ReactMethod
    override fun shareMiniProgram(request: ReadableMap, callback: Callback) {
        moduleImpl.shareMiniProgram(request, callback)
    }

    @ReactMethod
    override fun isWechatInstalled(callback: Callback) {
        moduleImpl.isWechatInstalled(callback)
    }

    @ReactMethod
    override fun requestPayment(request: ReadableMap, callback: Callback) {
        moduleImpl.requestPayment(request, callback)
    }

    @ReactMethod
    override fun requestSubscribeMessage(request: ReadableMap, callback: Callback) {
        moduleImpl.requestSubscribeMessage(request, callback)
    }

    @ReactMethod
    override fun launchMiniProgram(request: ReadableMap, callback: Callback) {
        moduleImpl.launchMiniProgram(request, callback)
    }

    @ReactMethod
    override fun openCustomerService(request: ReadableMap, callback: Callback) {
        moduleImpl.openCustomerService(request, callback)
    }

    override fun addListener(eventType: String) {

    }

    override fun removeListeners(count: Double) {

    }

    companion object {
        const val NAME = "Wechat"
    }
}